src/java/org/apache/commons/lang/reflect/FieldUtils.java

```diff
@@ -0,0 +1,605 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.lang.reflect;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+/**
+ * Utilities for working with fields by reflection. Adapted and refactored
+ * from the dormant [reflect] Commons sandbox component.
+ * <p>
+ * The ability is provided to break the scoping restrictions coded by the
+ * programmer. This can allow fields to be changed that shouldn't be. This
+ * facility should be used with care.
+ *
+ * @author Stephen Colebourne
+ * @author Matt Benson
+ * @since 2.5
+ * @version $Id$
+ */
+public class FieldUtils {
+
+    /**
+     * FieldUtils instances should NOT be constructed in standard programming.
+     * <p>
+     * This constructor is public to permit tools that require a JavaBean instance
+     * to operate.
+     */
+    public FieldUtils() {
+        super();
+    }
+
+    /**
+     * Gets an accessible <code>Field</code> by name repecting scope.
+     * Superclasses/interfaces will be considered.
+     *
+     * @param cls  the class to reflect, must not be null
+     * @param fieldName  the field name to obtain
+     * @return the Field object
+     * @throws IllegalArgumentException if the class or field name is null
+     */
+    public static Field getField(Class cls, String fieldName) {
+        Field field = getField(cls, fieldName, false);
+        MemberUtils.setAccessibleWorkaround(field);
+        return field;
+    }
+
+    /**
+     * Gets an accessible <code>Field</code> by name breaking scope
+     * if requested. Superclasses/interfaces will be considered.
+     *
+     * @param cls  the class to reflect, must not be null
+     * @param fieldName  the field name to obtain
+     * @param forceAccess  whether to break scope restrictions using the
+     *  <code>setAccessible</code> method. <code>False</code> will only
+     *  match public fields.
+     * @return the Field object
+     * @throws IllegalArgumentException if the class or field name is null
+     */
+    public static Field getField(final Class cls, String fieldName, boolean forceAccess) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 74-111 into a separate method.